hazelcast/src/main/java/com/hazelcast/internal/eviction/impl/comparator/LRUEvictionPolicyComparator.java

```diff
@@ -33,9 +33,10 @@
 
     @Override
     public int compare(EvictableEntryView e1, EvictableEntryView e2) {
-        int result = Long.compare(e1.getLastAccessTime(), e2.getLastAccessTime());
-        // if access times are same, we try to select oldest entry to evict
-        return result == 0 ? Long.compare(e1.getCreationTime(), e2.getCreationTime()) : result;
+        long time1 = Math.max(e1.getCreationTime(), e1.getLastAccessTime());
+        long time2 = Math.max(e2.getCreationTime(), e2.getLastAccessTime());
```

Comment:
Do we need these comparisons after the change in `AbstractRecordStore`? I don't know exactly where the last access times can be changed, so it's just a question to learn the cases that need them.

Reply:
This comparator is a common one used by multiple data structures, it is not specific to IMap, i have also fixed it here.

Reply:
I think it would be better if we also apply the change made in `AbstractRecordStore` to `AbstractCacheRecordStore` to have a convenience on these last access times.

Reply:
There is no issue in the context of icache. Setting last-access-time is a missing requirement for imap after 4.2 improvements. No need for others now.